docs/contributor/04-10-technical-design.md

```diff
@@ -146,21 +146,20 @@ For now, the following scenarios must be covered by this component:
 - Restart Pods with proxy sidecar when CNI config changes.
 - Restart Pods with proxy sidecar after an Istio version update.
 - Restart Pods with proxy sidecar when proxy resources change.
+- Restart Pods if they match predicates that IstioResourcesReconciliation component will specify (e.g. being up-to-date with an EnvoyFilter)
 
-#### IstioIngressGatewayReconciliation
+#### IstioResourcesReconciliation
 
-The IstioIngressGatewayReconciliation component is responsible for bringing Istio Ingress Gateway to the desired state.
+IstioResourcesReconciliation is a component responsible for applying resources dependent on Istio (e.g. VirtualService, EnvoyFilter) and making sure that the state of Istio service-mesh is configured according to those resources.
+To ensure the correct state, IstioResourcesReconciliation component provides predicates on per-resource basis, that are consumed by IstioIngressGatewayReconciliation and ProxySidecarReconciliation. The predicates specify when does a restart of the aforementioned components should happen.
```

Comment:
Do you think that it would be useful to have a simple diagram showing the relation between the components IstioResourcesReconciliation, IstioIngressGatewayReconciliation, ProxySidecarReconciliation and Istio Controller and where itnerfaces are used for communication? I always feel like diagrams are easier to consume than text. But we can also think about adding a diagram like this later, when we finalized the implementation and have stable names for functions. 😁

Reply:
I think that some sort of class/interaction/communication diagram make sense, but I would leave that for later as those diagrams still need to be adjusted according to https://github.com/kyma-project/istio/issues/188